### PR TITLE
Converted cache-tests to typescript

### PR DESCRIPTION
**Pull request by joshheyse**
Added generic typing to lib/cache/from-config
nim-tests got added after a make lint (--fix)

### CODE REVIEW

**Review by partouf — APPROVED** [review]

File: test/cache-tests.ts [hunk 1]
```diff
@@ -211,17 +213,17 @@ function setup() {
             }
             const result = S3FS[params.Key];
             if (result) {
-                callback(null, {Body: result});
+                callback(undefined, {Body: result});
```
Comment (mattgodbolt) [inline test/cache-tests.ts:216]: oh good spot

File: test/cache-tests.ts [hunk 2]
```diff
@@ -234,18 +236,18 @@ describe('S3 tests', () => {
...
-        throwS3Error = new Error('Some s3 error');
+        throwS3Error = new Error('Some s3 error') as AWSError;
         return cache
-            .get('doesntmatter', 'subsystem')
+            .get('doesntmatter')
             .should.eventually.contain({hit: false})
             .then(x => {
                 cache.stats().should.eql({hits: 0, puts: 0, gets: 1});
-                err.toString().should.equal('Error: Some s3 error');
+                err!.toString().should.equal('Error: Some s3 error');
```
Comment (mattgodbolt) [inline test/cache-tests.ts:250]: The GHA tests we run are saying "Forbidden non-null assertion" on this - I'm not sure we deliberately did anything here though. @jeremy-rifkin I've not seen `!` before - what's your thoughts (I know we have an `assert` around). I know this is in tests, so I'm less worried but in the interests of learning I'd like to know if `!` is good vs assert etc.

File: test/nim-tests.ts [hunk 3]
```diff
@@ -26,9 +26,9 @@ import path from 'path';
 
 import {unwrap} from '../lib/assert';
 import {NimCompiler} from '../lib/compilers/nim';
+import {LanguageKey} from '../types/languages.interfaces';
```
Comment (mattgodbolt) [inline test/nim-tests.ts:29]: this seems unrelated: did the autoformatter do this? :( we are still trying to get the formatting and pre-commit hooks super reliable...

**Review by mattgodbolt — APPROVED** [review]
LGTM! One question for @jeremy-rifkin but that's mostly for my own edification (and also for noise reduction on the static checks)

File: test/nim-tests.ts (hunk 3 above)
Comment (mattgodbolt) [reply test/nim-tests.ts:29]: Ah you said in your OG post, ignore, sorry!

File: test/cache-tests.ts (hunk 2 above)
Comment (jeremy-rifkin) [reply test/cache-tests.ts:250]: TLDR my thoughts are it can be ok but may as well `unwrap` so we get a runtime check for it.

Typescript's `!` can be quick and helpful when you really know something is non-null but of course it's better either find a better way to type things or use something like unwrap in case anything ever goes wrong.

Comment (mattgodbolt) [reply test/cache-tests.ts:250]: In fairness here this is a test where it'd "just fail" in that case. I think I understand now. I'll merge as-is though, and we can discuss more :)
